backend code_cloud_functions/main.js

```diff
@@ -6,44 +6,47 @@ const CLASS_LIKE = "Like";
 const CLASS_COMMENT = "Comment";
 const KEY_AUTHOR = "author";
 const KEY_POST = "post";
+const KEY_CREATED_AT = "createdAt";
 
 //Retrieve 20 posts with all the info needed to populate the timeline
 Parse.Cloud.define("getTimeline", async(request) => {
     let query = new Parse.Query(CLASS_POST);
     query.limit(20);
-    query.descending()
+    query.descending(KEY_CREATED_AT);
+    query.include(KEY_AUTHOR);
 
     const posts = await query.find();
-    let timeline = []
+    let reactions = []
 
     for (let i = 0; i < posts.length; i++) {
-        let object = posts[i];
+        let object = {};
         //Add extra useful info
-        object["isLiked"] = isPostLiked(object.id, request.user.id);
-        object["likesCount"] = getLikesCount(object.id);
-        object["commentsCount"] = getCommentCount(object.id);
-        timeline.push(object);
+        object["isLiked"] = await isPostLiked(posts[i], request.user);
+        object["likesCount"] = await getLikesCount(posts[i]);
+        object["commentsCount"] = await getCommentCount(posts[i]);
+        reactions.push(object);
     }
 
-    return timeline;
+    return {posts, reactions};
```

Comment:
You could consider returning a wrapper model that is a pair of:
```
List<Pair<Post, Reactions>>
```

Then, you wouldn't need to worry about keeping those lists ordered but instead would have 1 model object that contains both.  Or you could even add a variable/field to the "object"/dictionary have right now that is the "post"